runtime/lua/vim/lsp/util.lua

```diff
@@ -1672,6 +1672,51 @@ function M.symbols_to_items(symbols, bufnr)
   return _symbols_to_items(symbols, {}, bufnr or 0)
 end
 
+do
+  local foldlevels = {}
+
+  --- Applies a list of fold ranges to a certain buffer.
+  ---
+  --@param bufnr buffer id
+  --@param ranges list of `FoldingRange` objects
```

Comment:
LuaCATs annotations need to begin with `---` and can you mark both of these functions as private.